`parse_headers`の実装が良くない

**バグ説明1**

このコードがコンパイルエラーになる

```rust
use traq_bot_http::RequestParser;

fn main() {
    let parser = RequestParser::new("test");
    let headers = [
        ("Content-Type", "application/json"),
        ("X-TRAQ-BOT-TOKEN", "test"),
        ("X-TRAQ-BOT-EVENT", "PING"),
    ];
    let kind = parser.parse_headers(headers.into_iter());
    println!("{:?}", kind);
}
```

```
error[E0277]: the size for values of type `str` cannot be known at compilation time
  --> examples/test.rs:10:23
   |
10 |     let kind = parser.parse_headers(headers.into_iter());
   |                       ^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `Sized` is not implemented for `str`
note: required by a bound in `RequestParser::parse_headers`
  --> /Users/kh/Documents/trap/traq-bot-http-rs/src/parser.rs:64:33
   |
64 |     pub fn parse_headers<'a, H, K, V>(&self, headers: H) -> Result<EventKind, ParseError>
   |                                 ^ required by this bound in `RequestParser::parse_headers`

For more information about this error, try `rustc --explain E0277`.
```

**望まれる挙動1**

ここの`K`と`V`に`?Sized`を足す

https://github.com/H1rono/traq-bot-http-rs/blob/d777c40dd3cb8fafff66661d7380efcd299c0fce/src/parser.rs#L67-L68

**バグ説明2**
不要な`dbg!`がある

https://github.com/H1rono/traq-bot-http-rs/blob/d777c40dd3cb8fafff66661d7380efcd299c0fce/src/parser.rs#L81

**望まれる挙動2**
消す

--- a/src/parser.rs
+++ b/src/parser.rs
@@ -78,18 +78,18 @@ impl RequestParser {
                 Ok(k) => k,
                 Err(_) => continue,
             };
-            dbg!(k);
+            let v = from_utf8(v.as_ref());
             match k.to_lowercase().as_str() {
                 "content-type" => {
-                    let v = from_utf8(v.as_ref()).map_err(|_| ParseError::ReadContentTypeFailed)?;
+                    let v = v.map_err(|_| ParseError::ReadContentTypeFailed)?;
                     content_type = Some(v);
                 }
                 "x-traq-bot-token" => {
-                    let v = from_utf8(v.as_ref()).map_err(|_| ParseError::ReadBotTokenFailed)?;
+                    let v = v.map_err(|_| ParseError::ReadBotTokenFailed)?;
                     token = Some(v);
                 }
                 "x-traq-bot-event" => {
-                    let v = from_utf8(v.as_ref()).map_err(|_| ParseError::ReadBotEventFailed)?;
+                    let v = v.map_err(|_| ParseError::ReadBotEventFailed)?;
                     kind = Some(v);
                 }
                 _ => continue,

--- a/src/parser.rs
+++ b/src/parser.rs
@@ -64,8 +64,8 @@ impl RequestParser {
     pub fn parse_headers<'a, H, K, V>(&self, headers: H) -> Result<EventKind, ParseError>
     where
         H: Iterator<Item = (&'a K, &'a V)>,
-        K: AsRef<[u8]> + 'static,
-        V: AsRef<[u8]> + 'static,
+        K: AsRef<[u8]> + ?Sized + 'static,
+        V: AsRef<[u8]> + ?Sized + 'static,
     {
         // Content-Type: application/json
         let mut content_type = None;

